src/main/java/com/klarna/hiverunner/CommandShellEmulation.java

```diff
@@ -35,6 +53,25 @@ public String transformScript(String script) {
     }
   },
   BEELINE {
+
+    private final String runCommand = "!run";
+
+    @Override
+    public boolean isImportFileStatement(String statement) {
+      // case-sensitive
+      return statement.trim().startsWith(runCommand);
+    }
+
+    @Override
+    public File getImportFileFromStatement(String statement) {
+      // filename cannot contain whitespace
+      String[] tokens = statement.trim().split(" ");
+      if (tokens.length == 2) {
+        return new File(tokens[1]);
+      }
+      throw new IllegalArgumentException("Cannot get file to import from '" + statement + "'");
+    }
+
     @Override
     public String transformStatement(String statement) {
       return filterFullLineComments(statement);
```

Comment:
According to this http://sqlline.sourceforge.net/#sect_command_run any rows starting with '#' should be ignored